### ISSUE TITLE
RSX: Disgaea 3 to_comparaison_function(): Wrong comparaison function 0

### ISSUE THREAD

**Issue by raven02** [RPCS3/rpcs3#1944]
Trigger from decode.h 

```
struct registers_decoder<NV4097_SET_DEPTH_FUNC>
{
	static auto decode(u32 value) { return to_comparaison_function(value); }

	static void commit_rsx_state(rsx_state &state, comparaison_function &&decoded_values)
	{
		state.m_depth_func = decoded_values;
	}

	static std::string dump(comparaison_function &&decoded_values)
	{
		return "Depth: compare_function = " + print_comparaison_function(decoded_values);
	}
};
```

**Comment by raven02** [comment]
The game failed to boot now and affect all backend  with error

{rsx::thread} class fmt::exception thrown: rsx::to_comparaison_function(): Wrong comparaison function 0
(in file Emu\RSX\GCM.cpp:808)

**Comment by raven02** [comment]
rsx::thread} RSX: CELL_GCM_NV4097_SET_DEPTH_FUNC(0x29b) = 0x0
{rsx::thread} class fmt::exception thrown: rsx::to_comparaison_function(): Wrong comparaison function 0
(in file Emu\RSX\GCM.cpp:808)

**Comment by raven02** [comment]
Front face is even more interesting . it has value both 0x900 (valid) and 0x1D01 (Invalid)

{rsx::thread} RSX: CELL_GCM_NV4097_SET_CULL_FACE_ENABLE(0x60f) = 0x0
{rsx::thread} RSX: CELL_GCM_NV4097_SET_FRONT_FACE(0x60d) = 0x900
{rsx::thread} RSX: CELL_GCM_NV4097_SET_FRONT_FACE(0x60d) = 0x1d01

**Comment by kd-11** [comment]
Looks like defaults will be required for some state parameters. Some games are rather buggy.

**Comment by Nekotekina** [comment]
My guess: only one byte (8 bits) is read by RSX, other bits are ignored and defined in GCM constants for debug purposes.

**Comment by vlj** [comment]
@raven02 yes they are always parsed now. This way it's easier to know if a non canonical value is met and refire rsx decoding constant (for instance here we should mask with &0xff).

**Comment by raven02** [comment]
front face read 8 bits only (& 0xff) . but how about the 0 depth function ?

{rsx::thread} RSX: CELL_GCM_NV4097_SET_DEPTH_FUNC(0x29b) = 0x0

**Comment by raven02** [comment]
This message is not getting on the log console until I stop the emulator .

@Nekotekina Is it an expected behaviour ?

**Comment by Nekotekina** [comment]
Sort of, it waits for VBlank thread to stop. I'll try to fix it.

**Comment by raven02** [comment]
Thanks then we can resume for the game to continue on such bogus commands

**Comment by raven02** [comment]
fixed by #1982
